### ISSUE TITLE
Support partial class generation

### ISSUE THREAD

**Issue by davhdavh** [VocaDB/ResXFileCodeGenerator#5]
Please support generating partial classes.

Please support these 4 options: 

- IsPartial (to enable this feature) (default false)
- ChildClass (name of child class, default "")
- GenerateGetter (only used when ChildClass is set, default false)
- ChildClassIsPublic (only used when ChildClass is set, default false)

Example Pages/Index.resx (`IsPartial=true, ChildClass=""`):

```cs
namespace Web.Pages
{
    using System.Globalization;
    using System.Resources;

    internal static partial class Index  //internal and static from existing options
    {
         ...
    }
}
```

Example (`IsPartial=true, ChildClass="Resources"`):

```cs
namespace Web.Pages
{
    using System.Globalization;
    using System.Resources;

    internal static partial class Index  //internal and static from existing options
    {
         internal static class Resources { //static from existing options
            ...
         }
    }
}
```

Example (`IsPartial=true, ChildClass="Resources", GenerateGetter=true`):

```cs
namespace Web.Pages
{
    using System.Globalization;
    using System.Resources;

    internal static partial class Index  //internal and static from existing options
    {
         internal static IndexResources Resources { get; } = new();

         internal class IndexResources { //cannot be static
            ...
         }
    }
}
```

Example (`IsPartial=true, ChildClass="Resources", GenerateGetter=true, ChildClassIsPublic=true`):

```cs
namespace Web.Pages
{
    using System.Globalization;
    using System.Resources;

    internal static partial class Index  //internal and static from existing options
    {
         public static IndexResources Resources { get; } = new();

         public class IndexResources { //cannot be static
            ...
         }
    }
}
```

**Comment by ycanardeau** [comment]
It makes sense to implement the IsPartial option, but I don't see much benefit of implementing other three options for our use-case. Can you please explain a bit more clearly "why" you need these options? It would be better to fork the repo and/or write another source generator that fits for your use-case.

By the way, I would rename
- `IsPartial` to `PartialClass`
- `ChildClass` to `InnerClassName`
- `ChildClassIsPublic` to `PublicInnerClass`

Since the `StaticClass` option defaults to `true`, I think
```cs
internal partial class Index
```
in the first example should be
```cs
internal static partial class Index
```
?

And can we generate
```cs
internal static Resources Default { get; } = new();    // The name is fixed to Default, and cannot be changed.
```
instead of
```cs
internal static IndexResources Resources { get; } = new();
```
in the third example?

**Comment by davhdavh** [comment]
> Since the StaticClass option defaults to true, I think
> in the first example should be

Yes, you are correct, typo from me. I updated for all 4 examples.

>And can we generate

My thinking was it should be possible to use similar code in asp.net, razor pages, models and other files.

```
namespace Web.ASP.NET;
public partial class ErrorPage {
    protected static class Resources {
...
        public static string ErrorMessage => _r.GetString("ErrorMessage");
...
    }
}
```
In both .aspx and in .aspx.cs you can then use `Resources.ErrorMessage`.
This also works for standalone classes, controllers etc. (although private inner class make make more sense for those)

```
namespace Web.RazorPages;
public partial class ErrorModel 
{
   internal ErrorModelResources Resources { get; } = new();

   internal class ErrorModelResources 
   {
      public string ErrorMessage => _r.GetString("ErrorMessage");

   }
}
```
In view you can use `@Model.Resources.ErrorMessage` and in the model itself you can use `Resources.ErrorMessage `.

**Comment by ycanardeau** [comment]
Implemented in v3.1.0, thanks to @davhdavh.
